src/monattack.cpp

```diff
@@ -1650,6 +1650,76 @@ void mattack::leap(monster *z, int index)
     }
 }
 
+void mattack::impale(monster *z, int index)
+{
+    if( !z->can_act() ) {
+        return;
+    }
+    Creature *target = z->attack_target();
+    if( target == nullptr || rl_dist( z->pos(), target->pos() ) > 1 ) {
+        return;
+    }
+
+    player *foe = dynamic_cast< player* >( target );
+    bool seen = g->u.sees( *z );
+
+    z->moves -= 80;
+    bool uncanny = foe != nullptr && foe->uncanny_dodge();
+    if( uncanny || dodge_check(z, target) ){
```

Comment:
That reset timer thing was more about it retrying until it finds a chink in the armor.
Dodging the entire hit should have some sort of a reward, otherwise the zed will retry immediately and most likely hit. Either extra move cost for the zed (like with a proper hit) or timer reset, preferably the latter.